File: Algorithms/src/2126. Destroying Asteroids/Code/solution.py
class Solution:
    def asteroidsDestroyed(self, mass: int, asteroids: List[int]) -> bool:

        # Sort asteroids from smallest to largest
        asteroids.sort()

        # Current planet mass
        current_mass = mass

        # Try destroying each asteroid
        for asteroid in asteroids:

            # Planet is too small
            if current_mass < asteroid:
                return False

            # Gain asteroid mass
            current_mass += asteroid

        # All asteroids destroyed
        return True

File: Algorithms/src/2126. Destroying Asteroids/Code/solution.cpp
class Solution {
public:
    bool asteroidsDestroyed(int mass, vector<int>& asteroids) {
        // Sort asteroids from smallest to largest
        sort(asteroids.begin(), asteroids.end());

        // Use long long because mass can become very large
        long long currentMass = mass;

        // Try destroying asteroids one by one
        for (int asteroid : asteroids) {

            // If planet is too small, it gets destroyed
            if (currentMass < asteroid) {
                return false;
            }

            // Gain the asteroid's mass
            currentMass += asteroid;
        }

        // All asteroids were destroyed
        return true;
    }
};